ci: build C library as separate workflow job

tseaver opened the pull request.

tseaver committed:
ci: build C library as separate workflow job

tseaver committed:
ci: drop old monolithic presubmit workflow

parthea commented on .github/workflows/presubmit-fine-grained.yml:
```diff
@@ -0,0 +1,79 @@
+            - 3.10-dev
+        exclude:
+            - manylinux: manylinux1_x86_64
+              python: 3.10-dev
```
Could you add a note to explain the reason that this configuration is excluded?

tseaver replied:
4614f39

parthea reviewed (commented):
Added one observation

tseaver committed:
ci: explain excluded matrix entry

Per review request.

tseaver committed:
ci: try to wire up qemu FBO aarch64

tseaver committed:
Merge branch 'main' into decompose-linux-workflow-build

tseaver committed:
ci: push relevant new env vars down to wheel build

tseaver committed:
ci: drop pre-build of C extension

'setup.py' now relies on 'CRC32c_INSTALL_PREFIX' to dead-reckon to
include / library paths when building the wheel.

tseaver committed:
fix: typo

tseaver committed:
fix: use correct install prefix

tseaver committed:
ci: dead-reckon using installed prefix

tseaver committed:
ci: fix dead reckoning

tseaver committed:
fix: typo

tseaver committed:
ci: drop pre-build of extension for local linux build

tseaver committed:
ci: use container-relative install prefix

tseaver committed:
ci: allow for libs installed into 'lib64'

tseaver committed:
chore: grrr

tseaver committed:
chore: grrrrrrr

tseaver committed:
ci: weird speling for rpath

tseaver committed:
ci: attempt to cache built C library

tseaver committed:
fix: typo

tseaver committed:
ci: fix setting step output variable

tseaver committed:
ci: test cache lit for C library

tseaver committed:
ci: make docker setup conditional on cache miss
